rust/src/server.rs

```diff
@@ -58,7 +70,7 @@ where
         let server = self.server.clone();
 
         Box::pin(async move {
-            let req: HyperRequest<Body> = match req.headers().get("authorization") {
+            let (mut req, tag): (HyperRequest<Body>, tonic::codegen::http::HeaderValue) = match req.headers().get("authorization") {
```

Comment:
This type syntax is a real bummer

Reply:
Yeah, HyperRequest<Body> is needed because it needs to know the body type and HeaderValue is needed so it can properly be parsed.